ingress/ingress.go

```diff
@@ -337,33 +337,49 @@ func (c *ControlServer) handleIngressUpdate(oldObj interface{}, newObj interface
 }
 
 func (c *ControlServer) ingressChanged(old *netv1beta1.Ingress, new *netv1beta1.Ingress) bool {
-	if len(new.Spec.Rules) > 0 {
-		r0 := new.Spec.Rules[0]
-		hName := r0.Host
 
-		// If hostname changed, re-create
-		if hName != old.Spec.Rules[0].Host {
+	//first check top level changes like annotations
+	// try and get out early with a simple length check
+	if len(old.Annotations) != len(new.Annotations) {
+		return true
+	}
+	// check regular string annotations
+	for k, v := range new.Annotations {
+		if old.Annotations[k] != v && k != tyk.TemplateNameKey {
```

Comment:
I think this should be specifically checking Tyk-related annotations:
https://github.com/TykTechnologies/tyk-k8s/blob/1193b04e62692a9cccb5ec64a610c1aaf28f29a2/processor/proc.go#L14:L20

We don't really care if some other controller modified its annotations (e.g. cert-manager).